### PR TITLE
🧹 Adding logging to configureSendConfig

### PR DESCRIPTION
**Pull request by ravinagill15**
## In this PR:
- Added logging to `configureSendConfig` in `ua-devtools > src > oapp > config.ts`

### CODE REVIEW

File: packages/ua-devtools/src/oapp/config.ts [hunk 1]
```diff
@@ -240,84 +240,126 @@ export const configureReceiveLibraryTimeouts: OAppConfigurator = withOAppLogger(
...
+                        setConfigsByEndpointAndLibrary.getOrElse(from, () => new Map()).get(currentSendLibrary) ?? []
+                    const updatedConfigListCsv = updatedConfigList
+                        .map(
+                            (setConfigParam) =>
+                                '{configType: ' + setConfigParam.configType + ', config: setConfigParam.config}'
+                        )
+                        .join(', ')
+                    logger.verbose(
+                        `Set executor configuration ${updatedConfigListCsv} for ${formatOmniVector({ from, to })}`
+                    )
```
Comment (ravinagill15) [inline packages/ua-devtools/src/oapp/config.ts:307]: This is the change , along with the other log above (the rest of the function was just indented)

File: packages/ua-devtools/src/oapp/config.ts [hunk 2]
```diff
@@ -240,84 +240,126 @@ export const configureReceiveLibraryTimeouts: OAppConfigurator = withOAppLogger(
...
+                    const updatedConfigList =
+                        setConfigsByEndpointAndLibrary.getOrElse(from, () => new Map()).get(currentSendLibrary) ?? []
+                    const updatedConfigListCsv = updatedConfigList
+                        .map(
+                            (setConfigParam) =>
+                                '{configType: ' + setConfigParam.configType + ', config: setConfigParam.config}'
+                        )
+                        .join(', ')
+                    logger.verbose(
+                        `Set Uln configuration ${updatedConfigListCsv} for ${formatOmniVector({ from, to })}`
```
Comment (ravinagill15) [inline packages/ua-devtools/src/oapp/config.ts:349]: This is the change , along with the other log above (the rest of the function was just indented)

File: packages/ua-devtools/src/oapp/config.ts [hunk 3]
```diff
@@ -240,84 +240,126 @@ export const configureReceiveLibraryTimeouts: OAppConfigurator = withOAppLogger(
...
+                const hasExecutorConfig = await endpointSdk.hasAppExecutorConfig(
+                    from.address,
+                    currentSendLibrary,
+                    to.eid,
+                    config.sendConfig.executorConfig
                 )
+
+                logger.verbose(
+                    `Checked executor configuration for ${formatOmniVector({ from, to })}: ${printBoolean(hasExecutorConfig)}`
+                )
```
Comment (ravinagill15) [inline packages/ua-devtools/src/oapp/config.ts:281]: change 1

File: packages/ua-devtools/src/oapp/config.ts [hunk 4]
```diff
@@ -240,84 +240,126 @@ export const configureReceiveLibraryTimeouts: OAppConfigurator = withOAppLogger(
...
+                    const updatedConfigList =
+                        setConfigsByEndpointAndLibrary.getOrElse(from, () => new Map()).get(currentSendLibrary) ?? []
+                    const updatedConfigListCsv = updatedConfigList
+                        .map(
+                            (setConfigParam) =>
+                                '{configType: ' + setConfigParam.configType + ', config: ' + setConfigParam.config + '}'
+                        )
+                        .join(', ')
+                    logger.verbose(
+                        `Set executor configuration ${updatedConfigListCsv} for ${formatOmniVector({ from, to })}`
```
Comment (ravinagill15) [inline packages/ua-devtools/src/oapp/config.ts:306]: change 2

File: packages/ua-devtools/src/oapp/config.ts [hunk 5]
```diff
@@ -240,84 +240,126 @@ export const configureReceiveLibraryTimeouts: OAppConfigurator = withOAppLogger(
...
-                ])
-
-                // Updates map with new configs for that OApp and Send Library
-                const setConfigsByLibrary = setConfigsByEndpointAndLibrary.getOrElse(from, () => new Map())
-                const existingSetConfigs = setConfigsByLibrary.get(currentSendLibrary) ?? []
-                setConfigsByEndpointAndLibrary.set(
-                    from,
-                    setConfigsByLibrary.set(currentSendLibrary, [...existingSetConfigs, ...newSetConfigs])
+                logger.verbose(
+                    `Checked Uln configuration for ${formatOmniVector({ from, to })}: ${printBoolean(hasUlnConfig)}`
```
Comment (ravinagill15) [inline packages/ua-devtools/src/oapp/config.ts:324]: change 3

File: packages/ua-devtools/src/oapp/config.ts [hunk 6]
```diff
@@ -240,84 +240,126 @@ export const configureReceiveLibraryTimeouts: OAppConfigurator = withOAppLogger(
...
+                        setConfigsByEndpointAndLibrary.getOrElse(from, () => new Map()).get(currentSendLibrary) ?? []
+                    const updatedConfigListCsv = updatedConfigList
+                        .map(
+                            (setConfigParam) =>
+                                '{configType: ' + setConfigParam.configType + ', config: ' + setConfigParam.config + '}'
+                        )
+                        .join(', ')
+                    logger.verbose(
+                        `Set Uln configuration ${updatedConfigListCsv} for ${formatOmniVector({ from, to })}`
+                    )
```
Comment (ravinagill15) [inline packages/ua-devtools/src/oapp/config.ts:350]: change 4

File: packages/ua-devtools/src/oapp/config.ts [hunk 7]
```diff
@@ -240,84 +240,126 @@ export const configureReceiveLibraryTimeouts: OAppConfigurator = withOAppLogger(
...
 
-    // This function iterates over the map (OApp -> SendLibrary -> SetConfigParam[]) to execute setConfig
-    return buildOmniTransactions(setConfigsByEndpointAndLibrary, createSdk)
-}
+        // This function iterates over the map (OApp -> SendLibrary -> SetConfigParam[]) to execute setConfig
+        return buildOmniTransactions(setConfigsByEndpointAndLibrary, createSdk)
+    },
+    {
+        onStart: (logger) => logger.verbose(`Checking send configuration`),
+        onSuccess: (logger) => logger.verbose(`${printBoolean(true)} Checked send configuration`),
```
Comment (ravinagill15) [inline packages/ua-devtools/src/oapp/config.ts:360]: change 5

**Review by ravinagill15 — COMMENTED** [review]
The changes highlighted by github are mostly due to indentation -- i have commented below the actual changes

File: packages/ua-devtools/src/oapp/config.ts (hunk 7 above)
Comment (ravinagill15) [reply packages/ua-devtools/src/oapp/config.ts:360]: Noticed that when set to `verbose` this log doesn't appear in the test build run. When set to `info`, it appears.

Other logs within the actual logic appear, even with the verbose flag, so I can leave those as is, but should I change these `onStart` and `onSuccess` logs to use `info`?

File: packages/ua-devtools/src/oapp/config.ts (hunk 3 above)
Comment (ravinagill15) [reply packages/ua-devtools/src/oapp/config.ts:281]: Despite[ this test ](https://github.com/LayerZero-Labs/devtools/blob/aae5fa2cd7a206968e46857a4e592000d5377db3/tests/ua-devtools-evm-hardhat-test/test/oapp/config.test.ts#L757-L767) having a `config.sendConfig.executorConfig`, this log doesn't appear in the test build run. Perhaps I am checking the wrong test?

I would like to confirm this log and the `updatedConfigListCsv` log in the test build run before merging this PR

Comment (ravinagill15) [reply packages/ua-devtools/src/oapp/config.ts:281]: The above is resolved -- once I expose the test networks on localhost, deploy and wire them with `--log-level verbose`, I am able to see these logs

File: .changeset/lemon-brooms-watch.md [hunk 8]
```diff
@@ -0,0 +1,5 @@
+---
+"@layerzerolabs/ua-devtools": patch
+---
+
+Added logging to configureReceiveConfig
```
Comment (janjakubnanista) [inline .changeset/lemon-brooms-watch.md:5]: I think this might be a misrebase

**Review by janjakubnanista — CHANGES_REQUESTED** [review]

**Review by janjakubnanista — APPROVED** [review]

**Pull request merged**